runtime/src/main/java/org/corfudb/protocols/wireprotocol/LogData.java

```diff
@@ -76,9 +76,8 @@ public Object getPayload(CorfuRuntime runtime) {
                         ByteBuf copyBuf = Unpooled.wrappedBuffer(data);
                         final Object actualValue =
                                 Serializers.CORFU.deserialize(copyBuf, runtime);
-                        // TODO: Remove circular dependency on logEntry.
                         if (actualValue instanceof LogEntry) {
-                            ((LogEntry) actualValue).setEntry(this);
+                            ((LogEntry) actualValue).setGlobalAddress(getGlobalAddress());
                             ((LogEntry) actualValue).setRuntime(runtime);
```

Comment:
I think Its time to remove this too  `.setRuntime(runtime)`
#1983

Reply:
Sure, will do after this PR.